library-model/library-model-generator/src/main/java/com/uber/nullaway/libmodel/LibraryModelGenerator.java

```diff
@@ -197,6 +201,17 @@ public void visit(ClassOrInterfaceDeclaration cid, Void arg) {
                   this.isNullMarked = true;
                 }
               });
+      ImmutableMap<Integer, ImmutableSet<String>> genericParamAnnotationsMap =
+          getGenericTypeParameterNullableUpperBounds(cid);
+      /*
+       We insert a specialized MethodAnnotationsRecord object to store the generic type parameter nullability
+       information for the class by using an identifier that can never be an actual method name.
+      */
+      if (this.isNullMarked && !genericParamAnnotationsMap.isEmpty()) {
+        methodRecords.put(
+            parentName + ":" + GENERIC_TYPE_IDENTIFIER,
+            MethodAnnotationsRecord.create(ImmutableSet.of(), genericParamAnnotationsMap));
```

Comment:
At first glance this seems rather hacky.  Why are we reusing the `methodRecords` map and the `MethodAnnotationsRecord` class when the upper bound is on a type variable?  At least for this case, the type variable is declared on a class, not a method.  I'd prefer we just create a separate data structure for class or type variable records

Reply:
I agree, it is very hacky. I will use a separate data structure and work on a way to integrate it within the cache.